src/main/java/racingcar/common/type/Names.java

```diff
@@ -0,0 +1,28 @@
+package racingcar.common.type;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import racingcar.common.validator.NamesValidator;
+
+public class Names {
+    private final List<Name> nameList;
+
+    private Names(List<Name> nameList) {
+        NamesValidator.validate(nameList);
+        this.nameList = List.copyOf(nameList);
+    }
+
+    public static Names of(String[] names) {
+        var list = Arrays.stream(names)
+                .map(Name::of)  // String을 Name 객체로 변환
+                .collect(Collectors.toList());
+        return new Names(list);
+    }
```

Comment:
toList를 바로 사용하는 것은 어떨까요?
```suggestion
    public static Names of(String[] names) {
        var list = Arrays.stream(names)
                .map(Name::of)  // String을 Name 객체로 변환
                .toList();
        return new Names(list);
    }
```
물론 둘의 반환값은 차이가 있지만, 위 코드에서 `List.copyOf(nameList);` 를 사용해 불변을 보장하려 하셨다면 위 코드가 더 적절하다고 생각되요!

참고자료: https://binux.tistory.com/146

Reply:
감사합니다!! 그 차이점은 몰랐네요 배우고 갑니다!